File: 3sem/4lab/Banks.Console/Handlers/ShowClientsHandler.cs
using Banks.Entities;
using Banks.Services;

namespace Banks.Console.Handlers;

public class ShowClientsHandler : Handler
{
    public override void Handle(string request, CentralBank centralBank)
    {
        ArgumentNullException.ThrowIfNull(request);
        ArgumentNullException.ThrowIfNull(centralBank);

        if (request.ToLower() == "show clients")
        {
            IEnumerable<Client> clients = centralBank.Banks.SelectMany(b => b.Clients);

            foreach (Client client in clients)
                System.Console.WriteLine($" * Client '{client.Name} {client.Surname}' {client.Id}");

            return;
        }

        NextHandler?.Handle(request, centralBank);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Presentation/Handlers/ShowClientsHandler.java
package BanksRemake.Presentation.Handlers;

import BanksRemake.Domain.Entities.Client;
import BanksRemake.Domain.Exceptions.InvalidBankAccountOperationException;
import BanksRemake.Domain.Exceptions.InvalidBankOperationException;
import BanksRemake.Domain.Exceptions.InvalidDepositInterestRateSelectorException;
import BanksRemake.Domain.Exceptions.InvalidMoneyGapException;
import BanksRemake.Domain.Services.CentralBank;

import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class ShowClientsHandler extends HandlerBase {
    @Override
    public void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException {
        if (request.toLowerCase().equals("show clients")) {
            List<Client> clients = centralBank.getBanks().stream()
                    .flatMap(b -> b.getClients().stream())
                    .distinct()
                    .collect(Collectors.toList());

            for (Client client : clients) {
                System.out.println(String.format(" * Client '%s' %s", client.getName(), client.getId()));
            }

            return;
        }

        if (getNext() != null) {
            getNext().handle(request, centralBank, scanner);
        }
    }
}
